cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/spark/ETLSpark.java

```diff
@@ -114,11 +120,25 @@ public void beforeSubmit(SparkContext context) throws Exception {
       finishers.add(batchSink, sinkContext);
     }
 
+    Set<StageInfo> aggregators = phaseSpec.getPhase().getStagesOfType(BatchAggregator.PLUGIN_TYPE);
+    Integer numPartitions = null;
+    if (!aggregators.isEmpty()) {
+      String aggregatorName = aggregators.iterator().next().getName();
```

Comment:
or just
`Iterables.getOnlyElement(sparkTransforms)` will assert the size==1 internally.